core/cmd/local_client.go

```diff
@@ -83,19 +84,12 @@ func applicationLockDB(cfg config.GeneralConfig, db *sqlx.DB, lggr logger.Logger
 	case "advisorylock", "dual":
 		advisoryLock = pg.NewAdvisoryLock(db, cfg.AdvisoryLockID(), lggr, cfg.AdvisoryLockCheckInterval())
 		if err = advisoryLock.TakeAndHold(); err != nil {
-			return cleanup, errors.Wrap(err, "error acquiring lock")
+			return nil, errors.Wrap(err, "error acquiring lock")
```

Comment:
I think this was intentional for `dual` so that the first lock would be released by the caller even though the second wasn't acquired.